common/djangoapps/rocket_chat/api/v0/views.py

```diff
@@ -0,0 +1,76 @@
+import logging
+
+from rest_framework.authentication import SessionAuthentication
+from rest_framework.response import Response
+from rest_framework.views import APIView
+from rest_framework import permissions
+from rest_framework import status
+
+from opaque_keys.edx.keys import CourseKey
+
+from rocket_chat.utils import (
+    get_rocket_chat_settings,
+    initialize_api_rocket_chat,
+    create_user,
+    get_subscriptions_rids,
+)
+from .serializers import RocketChatCredentialsSerializer
+
+LOG = logging.getLogger(__name__)
+
+
+class RocketChatCredentials(APIView):
+
+    authentication_classes = (
+        SessionAuthentication,
+    )
+    permission_classes = (permissions.IsAuthenticated,)
+
+    def get(self, request, course_id):
+        """
+        This returns a Json with the  user credentials in order to use the rocketchat api methods outside of the server
+        """
+        user = request.user
+
+        course_key = CourseKey.from_string(course_id)
+
+        rocket_chat_settings = get_rocket_chat_settings()
+
+        api_rocket_chat = initialize_api_rocket_chat(rocket_chat_settings)
+
+        if api_rocket_chat:
+
+            user_info = api_rocket_chat.users_info(username=user.username)
+
+            try:
+                user_info = user_info.json()
+                if not user_info.get('success', False):
+                    create_user(api_rocket_chat, user, course_key)
+            except AttributeError:
+                LOG.error("Rocketchat API response with: %s", user_info)
+                return Response(status=status.HTTP_500_INTERNAL_SERVER_ERROR)
+
+            response = api_rocket_chat.users_create_token(
+                username=user.username)
+
+            try:
+                response = response.json()
+            except AttributeError:
+                return Response(status=status.HTTP_500_INTERNAL_SERVER_ERROR)
+
+            url_service = rocket_chat_settings.get('public_url_service', None)
+
+            data = response.get('data', {})
+
+            auth_token = data.get('authToken', None)
+            user_id = data.get('userId', None)
+
+            subscriptions_rids = get_subscriptions_rids(api_rocket_chat, auth_token, user_id, True)
+
+            serializer = RocketChatCredentialsSerializer(
+                data={"url_service": url_service, "auth_token": auth_token, "user_id": user_id, "room_ids": list(subscriptions_rids)}
+            )
+            serializer.is_valid()
+            return Response(serializer.data, status=status.HTTP_200_OK)
+
+        return Response(status=status.HTTP_500_INTERNAL_SERVER_ERROR)
```

Comment:
Please log the actual errors that is happening here, there are 3 possible cases for server 500 errors and the only error here sent to the log file is the user information, please send true reasons for these server 500 errors to the log file so that we actual know why there is a server 500 error.